parcels/field.py

```diff
@@ -1041,6 +1037,107 @@ def spatial_c_grid_interpolation2D(self, ti, z, y, x, time):
              + (-eta * U + (1-xsi) * V) * py[3]) / jac
         return (u, v)
 
+    def spatial_c_grid_interpolation3D_full(self, ti, z, y, x, time):
+        grid = self.U.grid
+        xi = int(grid.xdim / 2) - 1
+        yi = int(grid.ydim / 2) - 1
+        (xsi, eta, zet, xi, yi, zi) = self.U.search_indices(x, y, z, xi, yi, ti, time)
+
+        if grid.gtype in [GridCode.RectilinearSGrid, GridCode.RectilinearZGrid]:
+            px = np.array([grid.lon[xi], grid.lon[xi+1], grid.lon[xi+1], grid.lon[xi]])
+            py = np.array([grid.lat[yi], grid.lat[yi], grid.lat[yi+1], grid.lat[yi+1]])
+        else:
+            px = np.array([grid.lon[yi, xi], grid.lon[yi, xi+1], grid.lon[yi+1, xi+1], grid.lon[yi+1, xi]])
+            py = np.array([grid.lat[yi, xi], grid.lat[yi, xi+1], grid.lat[yi+1, xi+1], grid.lat[yi+1, xi]])
+
+        if grid.mesh == 'spherical':
+            px[0] = px[0]+360 if px[0] < x-225 else px[0]
+            px[0] = px[0]-360 if px[0] > x+225 else px[0]
```

Comment:
what's special about the `225` here?

Reply:
This is nothing new. It's as before how we try to find in which cell our particle is located. It was meant to make it work even for coarse mesh (we want to mesh the 360 shift if distance is more than 180 but less than 270, alias 90 degrees)

Reply:
OK, is fine then